deepchem/models/torch_models/IRV.py

```diff
@@ -0,0 +1,163 @@
+    https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2750043/
+    
+    The paper introduces the Influence Relevance Voter (IRV), a novel machine learning model 
+    designed for virtual high-throughput screening (vHTS).vHTS predicts the biological activity 
+    of chemical compounds using computational methods, reducing the need for expensive experimental
+    screening.
+    
+    The IRV model extends the k-Nearest Neighbors (kNN) algorithm by improving how neighbors 
+    influence predictions. Instead of treating all neighbors equally, IRV assigns each neighbor
+    a relevance score based on its similarity to the query compound.This similarity is calculated
+    using molecular fingerprint comparisons between the query compound and its neighbors,allowing
+    more relevant neighbors to have a greater impact on the prediction.
+    """
+
+    def __init__(self, n_tasks, K, penalty):
+        """
+        Parameters
+        ----------
+        n_tasks: int
+            Number of tasks
+        K: int
+            Number of nearest neighbours used in classification
+        penalty: float
+            Amount of penalty (l2 or l1 applied)
+        """
+        super(IRVLayer, self).__init__()
+        self.n_tasks = n_tasks
+        self.K = K
+        self.penalty = penalty
+
+        # Initialize weights and biases
+        self.V = nn.Parameter(torch.tensor([0.01, 1.], dtype=torch.float32))
+        self.W = nn.Parameter(torch.tensor([1., 1.], dtype=torch.float32))
+        self.b = nn.Parameter(torch.tensor([0.01], dtype=torch.float32))
+        self.b2 = nn.Parameter(torch.tensor([0.01], dtype=torch.float32))
+
+    def forward(self, inputs: torch.Tensor ) -> List[torch.Tensor]:
+        K = self.K
+        outputs = []
+        for count in range(self.n_tasks):
+            # Similarity values
+            similarity = inputs[:, 2 * K * count:(2 * K * count + K)]
+            # Labels for all top K similar samples
+            ys = inputs[:, (2 * K * count + K):2 * K * (count + 1)].int()
+
+            R = self.b + self.W[0] * similarity + self.W[1] * torch.arange(1, K + 1, dtype=torch.float32, device=device)
+            R = torch.sigmoid(R)
+            z = torch.sum(R * self.V[ys], dim=1) + self.b2
+            outputs.append(z.view(-1, 1))
+        outputs= torch.cat(outputs, dim=1)
+       
+        logits = []
+        predictions=outputs
+        outputs=[]
+        for task in range(self.n_tasks):
+            task_output = Slice(task, 1)(predictions)
+            sigmoid = torch.sigmoid(task_output)
+            logits.append(task_output)
+            outputs.append(sigmoid)
+        outputs = torch.stack(outputs, dim=1)
+        outputs2 = 1 - outputs
+        outputs = [
+            torch.cat([outputs2, outputs], dim=2),
+            logits[0] if len(logits) == 1 else torch.cat(logits, dim=1)
+        ]
+        return outputs
+
+class MultitaskIRVClassifier(TorchModel):
```

Comment:
Docstring, reference, type annotations

Reply:
Doc string and type annotations added. Reference paper is already present in the beginning. Not sure what to add.